### ISSUE TITLE
Must toggle :syntax on then :syntax off to use with :syntax off

### ISSUE THREAD

**Issue by Vaelatern** [vimwiki/vimwiki#423]
I understand this is a documented necessity, though I don't understand why.

Desire: be able to use vimwiki without syntax highlighting

Use case: I do not use syntax highlighting ever, to the extent that I no longer know what my code looks like with syntax highlighting on.

Current Workaround: Created 2 lines after line 190 of `plugin/vimwiki.vim` that say, in turn, `syntax on`, and then `syntax off`.

All functionaltiy works if I toggle through `:syntax on` then `:syntax off`, even if I do nothing between `on` and `off`.

Checked functionality works without ever toggling syntax.

- [x] Open default wiki index file.
- [x] Open default wiki index file in a new tab.
- [x] Select and open wiki index file.
- [x] Delete wiki file you are in.
- [x] Rename wiki file you are in.
- [ ] Follow/Create wiki link
- [ ] Split and follow/create wiki link
- [ ] Vertical split and follow/create wiki link
- [ ] Go back to parent(previous) wiki link - could not actually test
- [ ] Find next wiki link
- [ ] Find previous wiki link

- [ ] Vimwiki2HTML

**Comment by johnmarcampbell** [comment]
I accidentally unchecked (and then rechecked) one of the checkboxes in the previous post. So it shows that I edited it but I didn't actually make any changes.

**Comment by Vaelatern** [comment]
Still correct. I provide the checkbox format because it's easy for maintainers to note when things are fixed.

**Comment by johnmarcampbell** [comment]
Setting `syntax on` then `syntax off` will disable syntax highlighting whether the user wants it or not. The following snippet will do the toggling only if syntax highlighting is initially off:

```
if !exists("g:syntax_on")
  syntax on
  syntax off
endif
```
I put this on line 193 of `plugin/vimwiki.vim` and now it respects my highlighting settings whether they are initially on or off. Does this work for you? If so, we could start talking about getting it put in a PR.

**Comment by Vaelatern** [comment]
The real question is why vimwiki insists it needs `syntax on` when it will work without. What changes when syntax is toggled?

**Comment by ranebrown** [comment]
The regex patterns that are used to match links and other behavior aren't defined until the syntax file is loaded [here](https://github.com/vimwiki/vimwiki/blob/6787e0fb7e772b7470d7f8e48a4b653450ea0e98/syntax/vimwiki.vim#L17). It would probably be ok to make the regex patterns be defined in [vimwiki#vars#init](https://github.com/vimwiki/vimwiki/blob/6787e0fb7e772b7470d7f8e48a4b653450ea0e98/autoload/vimwiki/vars.vim#L784-L787) instead. The only issue I can think of would be slower performance when first loading the plugin when vim starts so some profiling would need done.

**Comment by ranebrown** [comment]
@Vaelatern If you want to test this out and provide the numbers we could make the change.

**Comment by Vaelatern** [comment]
So run `call vimwiki#vars#populate_syntax_vars(s:current_syntax)` ?

**Comment by ranebrown** [comment]
Yes. It should work to delete that line from the syntax file and move it to `vars.vim` at the end of the `vimwiki#vars#init` function and then do some profiling with `vim --startuptime time.log`

**Comment by Vaelatern** [comment]
I'm testing all features without making any changes, on a fresh install. Just to see if my vim setup works for this without the syntax hack or the vars init thing.

Then someone else will need to test, or I'll add a clean vim container, just for this test.

**Comment by Vaelatern** [comment]
At present only the splitting commands do not work without syntax on with my setup.

**Comment by Vaelatern** [comment]
```
Error detected while processing function vimwiki#vars#init:
line    3:
E121: Undefined variable: s:current_syntax
E116: Invalid arguments for function vimwiki#vars#populate_syntax_vars
```

**Comment by ranebrown** [comment]
Modify it to `call vimwiki#vars#populate_syntax_vars(vimwiki#vars#get_wikilocal('syntax'))`

**Comment by tinmarino** [comment]
@Vaelatern  what is the state of this issue ?

**Comment by Vaelatern** [comment]
I can follow links, but not split and follow.
